zetaclient/chains/solana/observer/observer.go

```diff
@@ -0,0 +1,118 @@
+package observer
+
+import (
+	"context"
+
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/zeta-chain/zetacore/pkg/bg"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	solanacontract "github.com/zeta-chain/zetacore/pkg/contract/solana"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/base"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/metrics"
+)
+
+var _ interfaces.ChainObserver = &Observer{}
```

Comment:
We can check for interface compliance without allocating memory, like this:
```suggestion
var _ interfaces.ChainObserver = (*Observer)(nil)
```

Reply:
https://github.com/zeta-chain/node/pull/2465/commits/2743f9dc1b608f81928dc99e067e9f65fcc6d716